src/main/java/de/undercouch/gradle/tasks/download/DownloadAction.java

```diff
@@ -148,7 +148,7 @@ private void execute(URL src, Project project) throws IOException {
         }
         
         //open URL connection
-        URLConnection conn = openConnection(src, timestamp, project);
+        URLConnection conn = openConnection(src, timestamp, destFile.length(), project);
```

Comment:
I'm not sure, but I don't think this will work if destFile does not exist. Did you test it?